system/common.cpp

```diff
@@ -381,9 +381,19 @@ long WEAK __syscall_sched_yield()
 	return 0;
 }
 
-long WEAK __syscall_set_thread_area(unsigned long tp)
+extern struct __pthread __dummy_thread;
+
+#if defined(__CHEERP__) && defined(__ASMJS__)
+[[cheerp::wasm]]
+#endif
+long __syscall_set_thread_area(unsigned long tp)
 {
+	#if defined(__CHEERP__) && defined(__ASMJS__)
+	__builtin_cheerp_set_thread_pointer(tp);
 	return 0;
+	#else
+	return ESRCH;
```

Comment:
why this return value? and in any case, it should be with a `-` sign when coming from the kernel